src/Sylius/Behat/Context/Domain/ProductContext.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Context\Domain;
+
+use Behat\Behat\Context\Context;
+use Sylius\Component\Core\Test\Services\SharedStorageInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+/**
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
+ */
+final class ProductContext implements Context
+{
+
+    /**
+     * @var RepositoryInterface
+     */
+    private $productRepository;
+
+    /**
+     * @var RepositoryInterface
+     */
+    private $variantRepository;
+
+    /**
+     * @var RepositoryInterface
+     */
+    private $reviewRepository;
+
+    /**
+     * @var SharedStorageInterface
+     */
+    private $sharedStorage;
```

Comment:
SharedContext should be first parameter, as it is first argument in a constructor.